lib/src/collection/impl/iterator.dart

```diff
@@ -51,12 +51,10 @@ class InterOpKtListIterator<T>
 
   @override
   void remove() {
-    // removing from list is wrong because is is a copy of the original list.
-    // remove should modify the underlying list, not the copy
-    // see how kotlin solved this:
-    // https://github.com/JetBrains/kotlin/blob/ba6da7c40a6cc502508faf6e04fa105b96bc7777/libraries/stdlib/js/src/kotlin/collections/InternalHashCodeMap.kt
-    throw UnimplementedError(
-        "remove() in not yet implemented. Please vote for https://github.com/passsy/dart_kollection/issues/5 for prioritization");
```

Comment:
Unfortunately, it's not that simple. As I wrote in this comment, the Iterator is operating on a copy of the list. Removing items there is not reflected in the actual list.

Reply:
Can you give me a scenario where this approach would fail?